Duplicati/Library/Backend/AzureBlob/AzureBlobWrapper.cs

```diff
@@ -37,141 +33,126 @@ namespace Duplicati.Library.Backend.AzureBlob
         {
-            BlobResultSegment segment;
-            try
-            {
-                segment = await _container.ListBlobsSegmentedAsync(null, false, ListDetails, null, null, null, _operationContext, cancelToken).ConfigureAwait(false);
-            }
-            catch (StorageException ex)
+            IAsyncEnumerable<BlobItem> blobs = GetBlobsWithExceptionHandling();
+
+            await foreach (var blobItem in blobs.WithCancellation(cancelToken).ConfigureAwait(false))
             {
-                if (ex.RequestInformation.HttpStatusCode == 404)
+                var blobName = Uri.UnescapeDataString(blobItem.Name.Replace("+", "%2B"));
+        
+                if (blobItem is { } bi)
                 {
-                    throw new FolderMissingException(ex);
+                    var lastModified = bi.Properties.LastModified?.UtcDateTime ?? DateTime.UtcNow;
+                    yield return new FileEntry(blobName, bi.Properties.ContentLength ?? 0, lastModified, lastModified);
                 }
-                throw;
-            }
-
-            foreach (var item in segment.Results)
-                yield return item;
-
-            while (segment.ContinuationToken != null)
-            {
-                segment = await _container.ListBlobsSegmentedAsync(null, false, ListDetails, null, segment.ContinuationToken, null, _operationContext, cancelToken).ConfigureAwait(false);
-
-                foreach (var item in segment.Results)
-                    yield return item;
             }
-        }
 
-        public virtual async IAsyncEnumerable<IFileEntry> ListContainerEntriesAsync([EnumeratorCancellation] CancellationToken cancelToken)
-        {
-            await foreach (var x in ListContainerBlobEntriesAsync(cancelToken).ConfigureAwait(false))
+            IAsyncEnumerable<BlobItem> GetBlobsWithExceptionHandling()
             {
-                var absolutePath = x.StorageUri.PrimaryUri.AbsolutePath;
-                var containerSegment = string.Concat("/", _containerName, "/");
-                var blobName = absolutePath.Substring(absolutePath.IndexOf(
-                    containerSegment, System.StringComparison.Ordinal) + containerSegment.Length);
-
-                var res = new FileEntry(Uri.UrlDecode(blobName.Replace("+", "%2B")));
                 try
                 {
-                    if (x is CloudBlockBlob cb)
-                    {
-                        var lastModified = new System.DateTime();
-                        if (cb.Properties.LastModified != null)
-                            lastModified = new System.DateTime(cb.Properties.LastModified.Value.Ticks, System.DateTimeKind.Utc);
-                        res = new FileEntry(res.Name, cb.Properties.Length, lastModified, lastModified);
-                    }
+                    // Note: AsyncPageable<BlobItem> return internally uses .ConfigureAwait(false).
+                    return _container.GetBlobsAsync(cancellationToken: cancelToken);
```

Comment:
If this is not awaited here, the exception handler will not catch the error, because the continuation will not be invoked until the task is awaited.